src/test/AlgoTests.hs

```diff
@@ -0,0 +1,93 @@
+module AlgoTests where
+
+import           Control.Monad                  ( void )
+import           Language.Haskell.Exts
+import           Test.Hspec
+import           Test.HUnit.Base                ( assertFailure )
+
+import           Algo
+
+-- | Tests for the "Algo" module.
+testAlgo :: Spec
+testAlgo = context "Algo" testCompareCons
```

Comment:
I would switch `context` and `describe` such that the tests are more consistent with the tests of the Free Compiler. But apart from this minor detail, the tests look good to me : )